sdk/core/azure-core/azure/core/pipeline/transport/_base.py

```diff
@@ -290,11 +294,18 @@ def _format_data(data):
     def format_parameters(self, params):
         # type: (Dict[str, str]) -> None
         """Format parameters into a valid query string.
+
         It's assumed all parameters have already been quoted as
         valid URL strings.
 
+        Passing your query parameters
+        directly through the `query` kwarg of
+        :class:`~azure.core.pipeline.transport.HttpRequest` automatically
+        calls this method for you.
+
         :param dict params: A dictionary of parameters.
         """
+        params = params or {}
```

Comment:
@lmazuel want to call this out. This way, if users pass in `params=None` or `query=None` from the kwarg in `HttpRequest`, we don't throw a `None` error and we tack on the `?` to the end of the url

Reply:
If params is not a dict, what's our expected behavior?

Reply:
params is typed as a `Dict[str, str]`. i'm not quite sure about the use case of users passing in`None` for query`, but I added this bc @lmazuel wanted to make sure if query is `None`, we add the `?` at the end of the url

Reply:
You should be able to pass in a number or a boolean query parameter value, no?

Reply:
that's true, let me change this to `Dict[str, Any]`